### DISCUSSION TITLE
How to trigger some event after model is updated in prisma?

### DISCUSSION THREAD

**Discussion by darkodavidovic** [prisma/prisma#15982] [Q&A] [answered]
Hi... Is any possible way to run some middleware or event after we update some model in prisma?
I want to fire the publish method after user is created or deleted so i can collect new data and send it over the subscriptions.
I tried using prisma.$use and setTimeOut, but i wonder is there any more elegant way except to hack arround? I have REST API and GraphQL and i dont want to place everywhere in my resolvers or services my publish method. I want to trigger the event on prisma method. Code i tried is:
```
prisma.$use(async (params, next) => {
      if (params.model === 'User' && params.action === 'delete') {
	      // code to publish on delete
	      setTimeout(async () => {
		      const users = await prisma.user.findMany()
                      pubsub.publish('USERS_ARRAY', users)
	      }, 10);
      }
      return await next(params)
})
```

**Comment by darkodavidovic** [comment, marked as answer]
I fixed it using this setup.

```
prisma.$use(async (params, next) => {
      if (params.model === 'User' && params.action === 'delete') {
	      // code to publish on delete
	      const results = await next(params)
	      // runs after if result is positive (need to be if below)
	      const users = await prisma.user.findMany()
              pubsub.publish('USERS_ARRAY', users)
              return results
      }
      // return for any other param.action and model except delete
      return await next(params)
})
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
